improver/nbhood/use_nbhood.py

```diff
@@ -408,17 +408,10 @@ def process(self, cube):
         yname = cube.coord(axis="y").name()
         xname = cube.coord(axis="x").name()
 
-        renormalize = True
-        if self.weights.shape == cube.shape:
-            weights = self.renormalize_weights(cube)
-            renormalize = False
-
         # Loop over any extra dimensions
+        weights = self.weights.data
         cubelist = iris.cube.CubeList([])
         for slice_3d in cube.slices([self.coord_masked, yname, xname]):
-            if renormalize:
-                weights = self.renormalize_weights(slice_3d)
```

Comment:
I think that as we delete this code we need to make sure that some of the edge cases that they cover are covered by a higher level unit test. I think we could get rid of this plugin and make the code a method on the ApplyNeighbourhoodProcessingWithAMask plugin. It would take a bit more refactoring but would get rid of complexity we don't need. We will need to change the nbhood_iterate_with_mask cli too. I'm happy to make all these changes if we agree they are sensible.

Reply:
> I think we could get rid of this plugin and make the code a method on the ApplyNeighbourhoodProcessingWithAMask plugin.

This makes a lot of sense 👍

> We will need to change the nbhood_iterate_with_mask cli too.

Already done in this PR.

Reply:
OK, I reverted the last commit that removed this (which brings back the `renormalize_weights` method to prevent tests from failing).

Reply:
For the record: after some investigation it turns out that `numpy.ma.average` (which is called by `iris.analysis.MEAN`) already does weight renormalization internally (https://github.com/numpy/numpy/blob/6ee4917/numpy/ma/extras.py#L625-L629), so there's no need to do it separately and safe to remove it from here.